src/control/server/mgmt_system.go

```diff
@@ -135,23 +169,9 @@ func (svc *mgmtSvc) localInstances(inRanks []uint32) ([]*IOServerInstance, error
 	return localInstances, nil
 }
 
-// PrepShutdown implements the method defined for the Management Service.
-//
-// Prepare data-plane instance managed by control-plane for a controlled shutdown,
-// identified by unique rank.
-//
-// Iterate over instances, issuing PrepShutdown dRPCs and record results.
-// Return error in addition to response if any instance requests not successful
-// so retries can be performed at sender.
-func (svc *mgmtSvc) PrepShutdownRanks(parent context.Context, req *mgmtpb.RanksReq) (*mgmtpb.RanksResp, error) {
-	if req == nil {
-		return nil, errors.New("nil request")
-	}
-	if len(req.GetRanks()) == 0 {
-		return nil, errors.New("no ranks specified in request")
-	}
-	svc.log.Debugf("MgmtSvc.PrepShutdownRanks dispatch, req:%+v\n", *req)
-
+// drpcOnLocalRanks iterates over local instances issuing dRPC requests
+// concurrently and returning system member results.
+func (svc *mgmtSvc) drpcOnLocalRanks(parent context.Context, req *mgmtpb.RanksReq, method int32) ([]*system.MemberResult, error) {
```

Comment:
Hmm. We should make the drpc methods a type alias rather than using bare int32. Not in this PR, but I noticed it elsewhere in the PR and I'd hate to perpetuate it. Using a type alias would make it harder to supply unknown values and would allow us to implement a stringer for the method names.

Reply:
yes . wanted to do that , but thought probably better in a follow-on

Reply:
addressing in https://jira.hpdd.intel.com/browse/DAOS-3413